ara/core/result.h

```diff
@@ -0,0 +1,382 @@
+#ifndef ARA_CORE_RESULT_H_
+#define ARA_CORE_RESULT_H_
+
+#include <type_traits>
+#include <boost/variant.hpp>
+#include "error_code.h"
+
+namespace ara {
+namespace core {
+
+template <typename T, typename E = ErrorCode>
+class Result final
+{
+    boost::variant<T, E> _data;
+
+public:
+    using value_type = T;
```

Comment:
What is the point of those aliases, if you keep on using `T` and `E` in definitions? :P

Reply:
it still used in function names in docs so i found it much confusing if suddenly i started using it in function bodies

Reply:
done